### PR TITLE
Image Customizer: Change additionalFiles to a list.

### PR DESCRIPTION
**Pull request by cwize1**
Change `additionalFiles` from a map to a list. This will ensure that the copy operation has a fixed and consistent ordering, instead of being randomized by the map's hash algorithm.

In addition, add a feature that allows the content of additional files to be specified inline with in the YAML config.

---

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Updated UTs.

### CODE REVIEW

File: toolkit/tools/imagecustomizerapi/additionalfile_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	additionalFiles := AdditionalFileList{
+		{
+			Destination: "",
+			Source:      "a.txt",
+		},
+	}
+	err := additionalFiles.IsValid()
+	assert.Error(t, err)
+	assert.ErrorContains(t, err, "invalid value at index 0")
+	assert.ErrorContains(t, err, "detination path must not be empty")
```
Comment (romoh) [inline toolkit/tools/imagecustomizerapi/additionalfile_test.go:23]: nit - typo `destination`

File: toolkit/tools/imagecustomizer/docs/configuration.md [hunk 2]
```diff
@@ -160,9 +162,11 @@ os:
         - [remove](#remove-string)
         - [updateLists](#updatelists-string)
         - [update](#update-string)
-    - [additionalFiles](#additionalfiles-mapstring-fileconfig)
-      - [fileConfig type](#fileconfig-type)
-        - [path](#fileconfig-path)
+    - [additionalFiles](#os-additionalfiles)
+      - [additionalFile type](#additionalfile-type)
```
Comment (romoh) [inline toolkit/tools/imagecustomizer/docs/configuration.md:166]: We should also update `additionalDirs` to follow same `source` `destination` terminology

Comment (cwize1) [reply toolkit/tools/imagecustomizer/docs/configuration.md:166]: Ok, I've opened a work item.

File: toolkit/tools/imagecustomizerapi/additionalfile_test.go (hunk 1 above)
Comment (cwize1) [reply toolkit/tools/imagecustomizerapi/additionalfile_test.go]: Fixed.

File: toolkit/tools/internal/safechroot/safechroot.go [hunk 3]
```diff
@@ -30,7 +30,13 @@ const BindMountPointFlags = unix.MS_BIND | unix.MS_MGC_VAL
 
 // FileToCopy represents a file to copy into a chroot using AddFiles. Dest is relative to the chroot directory.
 type FileToCopy struct {
-	Src         string
+	// The source file path.
```
Comment (dmcilvaney) [inline toolkit/tools/internal/safechroot/safechroot.go:33]: So, we have the old and new config structs  (json vs. yaml). We tend to use the json versions of those structs in the imager code a bunch. I'm not sure what feels least odd here, why re-implement the same data structure twice? But... I believe we also generally don't directly import the image customizer types outside the customizer tool...

(No real opinion here, just feels a bit odd somehow. Probably just a consequence of old design choices we live with now)

**Review by dmcilvaney — APPROVED** [review]

**Review by romoh — APPROVED** [review]

**Pull request merged**